angular_analyzer_plugin/test/angular_driver_test.dart

```
@@ -299,6 +301,348 @@ class BuildStandardHtmlComponentsTest extends AbstractAngularTest {
+      custom_events:
+        strEventImplicitCore:
+          type: String
+        strEventExplicitCore:
+          type: String
+          path: 'dart:core'
+        boolEventImplicitCore:
+          type: bool
+        boolEventExplicitCore:
+          type: bool
+          path: 'dart:core'
+''', null);
+
+    super.setUp();
+
+    final html = await angularDriver.getStandardHtml();
+    // validate
+    expect(html, isNotNull);
+    expect(html.customEvents, isNotNull);
+    expect(html.customEvents, hasLength(4));
+    {
+      final event = html.customEvents['strEventImplicitCore'];
+      expect(event, isNotNull);
+      expect(event.getter, isNull);
+      expect(event.eventType, isNotNull);
+      expect(event.eventType.toString(), 'String');
+      expect(
+          event.eventType.element.source.fullName, '/sdk/lib/core/core.dart');
+    }
+    {
+      final event = html.customEvents['strEventExplicitCore'];
+      expect(event, isNotNull);
+      expect(event.getter, isNull);
+      expect(event.eventType, isNotNull);
+      expect(event.eventType.toString(), 'String');
+      expect(
+          event.eventType.element.source.fullName, '/sdk/lib/core/core.dart');
+    }
+    {
+      final event = html.customEvents['boolEventImplicitCore'];
+      expect(event, isNotNull);
+      expect(event.getter, isNull);
+      expect(event.eventType, isNotNull);
+      expect(event.eventType.toString(), 'bool');
+      expect(
+          event.eventType.element.source.fullName, '/sdk/lib/core/core.dart');
+    }
+    {
+      final event = html.customEvents['boolEventExplicitCore'];
+      expect(event, isNotNull);
+      expect(event.getter, isNull);
+      expect(event.eventType, isNotNull);
+      expect(event.eventType.toString(), 'bool');
+      expect(
+          event.eventType.element.source.fullName, '/sdk/lib/core/core.dart');
+    }
+  }
+
+  // ignore: non_constant_identifier_names
+  Future test_customEvents_noSource_dynamic() async {
```

So if a custom event's source doesn't exist (whether the file or definition is missing), do we still allow it to be used as dynamic without warning?

---

Unfortunately for now I don't have the infrastructure plumbed up to report erroneous configs. That will have to come later...for now it will fail silently. :(

---

Fail silently as in treat an event with that name as an error, or treat it as dynamic?

---

Ah, yes, sorry, the latter.